src/Core/IdentityServer/CustomTokenRequestValidator.cs

```diff
@@ -100,6 +83,38 @@ protected async override Task<(User, bool)> ValidateContextAsync(CustomTokenRequ
                     context.Result.ValidatedRequest.ClientClaims.Add(claim);
                 }
             }
+
+            if (context.Result.CustomResponse == null || user.MasterPassword != null)
+            {
+                return;
+            }
+
+            // KeyConnector responses below
+
+            if (context.Result.ValidatedRequest.GrantType == "client_credentials")
+            {
+                // Apikey login
+                if (user.UsesKeyConnector) {
+                    // KeyConnectorUrl is configured in the CLI client, just disable master password reset    
+                    context.Result.CustomResponse["ResetMasterPassword"] = false;
+                }
+                return;
```

Comment:
I just restructured this section to make sure that apikey logins always hit this early return, even if they're not using Key Connector. Otherwise they'll continue to the SSO section below and hit nulls.